airflow-core/src/airflow/api_fastapi/execution_api/datamodels/variable.py

```diff
@@ -34,3 +34,9 @@ class VariablePostBody(StrictBaseModel):
 
     value: str | None = Field(alias="val")
     description: str | None = Field(default=None)
+
+
+class VariableDeleteBody(StrictBaseModel):
+    """Response body schema for after deleting variables."""
+
+    delete_count: int
```

Comment:
```suggestion
    count: int
```

Reply:
Although in theory, you could just sent `True` or `False` too since it is 0/1

Reply:
I am also ok if we don't want to do that and instead raise an error in API Server when count is 0.

So we have different options

Reply:
Delete should be idempotent - "ensure this is deleted", so an error should be reserved for db issues, or "found in secrets backend, can't delete" etc

Reply:
Hmm yeah i agree on that too. We shouldn't be raising an error in the api server for this, its either delete if found or stay silent. Same behaviour as earlier.

> so an error should be reserved for db issues, or "found in secrets backend, can't delete" etc

Right yes, but deletion from the secrets backend is not something we support.